api/src/pcapi/routes/backoffice/pivots/contexts/base.py

```diff
@@ -60,4 +69,36 @@ def update_pivot(cls, form: forms.EditPivotForm, pivot_id: int) -> bool:
 
     @classmethod
     def delete_pivot(cls, pivot_id: int) -> bool:
-        raise NotImplementedError()
+        """
+        Common implementation except for Allocine for which deleting a pivot is not allowed.
+        """
+        pivot_name = cls.pivot_name()
+        pivot_model = cls.pivot_class()
+        pivot = pivot_model.query.get_or_404(pivot_id)
+        cinema_provider_pivot = pivot.cinemaProviderPivot
+        assert cinema_provider_pivot  # helps mypy
+
+        venue_provider = (
+            providers_models.VenueProvider.query.options(sa.orm.joinedload(providers_models.VenueProvider.venue))
+            .filter_by(venueId=cinema_provider_pivot.venueId, providerId=cinema_provider_pivot.providerId)
+            .one_or_none()
+        )
+
+        if venue_provider:
+            flash(
+                f"Ce lieu est toujours synchronisé avec {pivot_name}, vous ne pouvez pas supprimer ce pivot",
+                "warning",
+            )
+            return False
```

Comment:
Je lis dans le ticket : « lorsque je clique une pop in indique “le pivot supprimé engendrera une supression de la synchro” ». Or dans l'implémentation que tu as factorisée (youpi !), on refuse de supprimer s'il existe une synchro. As-tu vu avec @elizoturner si, justement, le but ne serait pas de permettre de supprimer aussi le `venue_provider` pour supprimer la synchro ? C'est ce que je comprends dans le ticket.

Reply:
ah c'est pas ce que j'ai compris du ticket. mais je re-vérifie avec Elisa 👍